sys/src/9/port/sysfile.c

```diff
@@ -827,13 +825,11 @@ write(int fd, void *p, int32_t n, int64_t off)
 	n = 0;
 	c = fdtochan(fd, OWRITE, 1, 1);
 
-	if(off == -1LL)
-		sequential = 1;
-	else
-		sequential = 0;
+	if(off != -1LL)
+		dopwrite = 1;
 
 	if(waserror()) {
-		if(sequential){
+		if(dopwrite){
```

Comment:
if (!doprwrite)

`doprwrite==1` means that you are using `pwrite` semantics. The old `sequential` meant that you were doing `write` semantics.